src/datasets/utils/file_utils.py

```diff
@@ -1146,7 +1146,7 @@ def _prepare_single_hop_path_and_storage_options(
     Storage options are formatted in the form {protocol: storage_options_for_protocol}
     """
     token = None if download_config is None else download_config.token
-    if urlpath.startswith(config.HF_ENDPOINT) and "/resolve/" in urlpath:
+    if urlpath.startswith(config.HF_ENDPOINT):
```

Comment:
not a big deal but you can revert this no ? if `/resolve/` is not present, the URL points to a web page on HF or an API endpoint that can't be converted to a `hf://` link

Reply:
Thanks for pointing this out.

I wrongly assumed that all and HTTP URL was either convertible to hf:// or a non-HF URL.

Now you make me realize that there is a third option: an HF URL not convertible to hf://
- I think in this case we should pass HF authentication headers and HF user-agent

Do you agree, @lhoestq?

Reply:
for urls non convertible to hf:// we can let the user pass HTTP storage options IMO to keep things simple

Reply:
Done.